sdk/webpubsub/Azure.ResourceManager.WebPubSub/src/autorest.md

```diff
@@ -57,24 +57,40 @@ override-operation-name:
   WebPubSub_CheckNameAvailability: CheckWebPubSubNameAvailability
 
 rename-mapping:
+  WebPubSubResource: WebPubSub
+  SharedPrivateLinkResource: WebPubSubSharedPrivateLink
   RegenerateKeyParameters: WebPubSubRegenerateKeyContent
+  ClientConnectionCountRule.type: ThrottleType
+  ClientTrafficControlRule.type: TrafficThrottleType
+  EventListenerFilter.type: EventListenerFilterType
+  EventListenerEndpoint.type: EventListenerEndpointType
+  CustomCertificate: WebPubSubCustomCertificate
+  CustomDomain: WebPubSubCustomDomain
+  Replica: WebPubSubReplica
+  EventHubEndpoint: WebPubSubEventHubEndpoint
+  EventListener: WebPubSubEventListener
+  IPRule: PublicTrafficIPRule
+  ServiceKind: WebPubSubServiceKind
+
+request-path-to-resource-name:
+  /subscriptions/{subscriptionId}/resourceGroups/{resourceGroupName}/providers/Microsoft.SignalRService/webPubSub/{resourceName}/sharedPrivateLinkResources/{sharedPrivateLinkResourceName}: WebPubSubSharedPrivateLink
 
 directive:
   - rename-model:
       from: PrivateLinkResource
       to: WebPubSubPrivateLink
-  - rename-model:
-      from: SharedPrivateLinkResource
-      to: WebPubSubSharedPrivateLink
+#   - rename-model:
+#       from: SharedPrivateLinkResource
+#       to: WebPubSubSharedPrivateLink
   - rename-model:
       from: NameAvailability
       to: WebPubSubNameAvailability
   - rename-model:
       from: NameAvailabilityParameters
       to: WebPubSubNameAvailabilityParameters
-  - rename-model:
-      from: WebPubSubResource
-      to: WebPubSub
```

Comment:
I would suggest we change all of these rename-model to rename-mapping.
This could also help when you have to add clientNames to typespec if everything is in the same place.

Reply:
Rewrote all the renaming part.